plugin-server/src/worker/ingestion/message-formatter.ts

```diff
@@ -0,0 +1,271 @@
+import { PostIngestionEvent, Team } from '../../types'
+import { cloneObject, getPropertyValueByPath } from '../../utils/utils'
+
+// Sync with .../api/person.py and .../lib/constants.tsx
+const PERSON_DEFAULT_DISPLAY_NAME_PROPERTIES = ['email', 'Email', 'name', 'Name', 'username', 'Username', 'UserName']
+
+const TOKENS_REGEX_BRACKETS_INCLUDED = /(?<!\\){{?(.*?)(?<!\\)}}?/g
+const TOKENS_REGEX_DOUBLE_BRACKETS_EXCLUDED = /(?<=(?<!\\){{)(.*?)(?=(?<!\\)}})/g
+
+export type MessageFormatterOptions = {
+    event: PostIngestionEvent
+    team: Team
+    siteUrl: string
+    sourcePath: string
+    sourceName: string
+}
+
+export class MessageFormatter {
```

Comment:
Is this used anywhere? I couldn't find any reference to `MessageFormatter` from anywhere else than its own tests... 🤔

Reply:
It was only there for me to write a bunch of tests to check that all the expected formatting still works. Made tricky by the inability to generate hog realtime in the plugin server though so I'll likely just remove the tests for now...